packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -104,8 +105,28 @@ export type SdkType =
 export interface SdkBuiltInType extends SdkTypeBase {
   kind: SdkBuiltInKinds;
   encode: string;
+  name: string;
```

Comment:
Is the `name` guaranteed to be same as the name in typespec, for all subclass of `SdkTypeBase`? Or it only be same if under `SdkBuiltInType`?

If latter, should we just call it `tspName`?

Reply:
I think it might be necessary to have a tspName in the future maybe considering there might be name changes such `@clientName`.
But here we are following the same rule for models. In details:
if this is a type defined somewhere else from the std library, this name represents the `library name`, which is a name modified by various decorators. If this is a type defined in the std library, this is just `type.name`.

Reply:
Probably not going to block the PR. Just want to check if we've considered some corner cases.

Could somebody do this? (it is valid in playground)

```ts
@friendlyName("url")
scalar uri extends string;

model Resource {
  prop1: uri;
}
```

Reply:
I do not think any one needs to ever remember this
> if this is a type defined somewhere else from the std library, this name represents the library name, which is a name modified by various decorators. If this is a type defined in the std library, this is just type.name.

I am saying this because I am doing this in the code, but there should not be such a rule because no one could change the name of a type from the std library, I did not call the `getLibraryName` when constructing them. I could do that but I think it is meaningless because no one could change it.

I could align the implementation so that there is no such rule.

Reply:
```
@@friendlyName(TypeSpec.duration, "dur");

@encode(DurationKnownEncoding.seconds, float32)
scalar myDuration extends duration;
```

![image](https://github.com/Azure/typespec-azure/assets/53292327/f3a6dc4e-30b7-42e1-9237-4a8157a41b69)

Reply:
What I mean is, give the behavior of this `name`, it would be much safer to dev to just use `__raw.name` if we really want to know the typespec name of the type.

I am OK with your PR. Just saying given the complication of this `type.name`, I won't really using it to check typespec name.

Reply:
`raw` is a workaround, we could use it for now, but we cannot use it forever, therefore we definitely need something else to represent the original name for a type.
If we could even add decorators to the std types, I will make a change to always call `getLibraryName` to be align. It may not make sense but i think we must align because no one could remember those complicated rules.

Reply:
per offline discussion, we should be able to identify types from template using `crossLanguageDefinitionId`, therefore we do not need to have the original name?

Reply:
Oh - after I checked my code, there is no such rule in the first place. All the `name` is constructed by `getLibraryName` even for those primitive types.